sematic/types/registry.py

```diff
@@ -238,6 +242,24 @@ def _register_to_json_encodable_summary(
     return _register_to_json_encodable_summary
 
 
+def register_type_assertion(
+    type_assertion: TypeAssertionCallable,
+) -> TypeAssertionCallable:
```

Comment:
Is this really necessary? Can these assertions be implemented in the type-checking logic instead?

Reply:
refactored to use an inline import